diff --git a/.DS_Store b/.DS_Store
new file mode 100644
index 0000000..e69de29
diff --git a/src/AutoField.tsx b/src/AutoField.tsx
index 79bbd02..ff2470d 100644
--- a/src/AutoField.tsx
+++ b/src/AutoField.tsx
@@ -2,15 +2,18 @@ import invariant from 'invariant';
 import { createAutoField } from 'uniforms';
 export { AutoFieldProps } from 'uniforms';

+import TextField from './TextField';
 import NumField from './NumField';
+
 import BoolField from './BoolField';
 import DateField from './DateField';
-import TextField from './TextField';
+
 / import ListField from './ListField';
 import RadioField from './RadioField';
 / import SelectField from './SelectField';

 const AutoField = createAutoField((props) => {
+
   if (props.allowedValues) {
     return RadioField;
     / return props.checkboxes && props.fieldType !== Array
@@ -32,7 +35,6 @@ const AutoField = createAutoField((props) => {
     case String:
       return TextField;
   }
-
   return invariant(false, 'Unsupported field type: %s', props.fieldType);
 });

diff --git a/src/MultiLineField.tsx b/src/MultiLineField.tsx
new file mode 100644
index 0000000..dcd4773
--- /dev/null
+++ b/src/MultiLineField.tsx
@@ -0,0 +1,72 @@
+import React from 'react';
+import { View } from 'react-native';
+import { FieldProps, connectField } from 'uniforms';
+import { TextInput, HelperText } from 'react-native-paper';
+
+export type MultiLineFieldProps = FieldProps<
+  string,
+  React.ComponentProps<typeof TextInput>,
+  {
+    helperText?: string;
+    inputRef?: any;
+  }
+>;
+
+function Text({
+  disabled,
+  error,
+  errorMessage,
+  helperText,
+  inputRef,
+  label,
+  name,
+  onChange,
+  placeholder,
+  readOnly,
+  showInlineError,
+  value = '',
+  ...props
+}: MultiLineFieldProps) {
+  let formHelperText = (error && showInlineError && errorMessage) || helperText;
+
+  let helperProps: any = {
+    type: showInlineError && error ? 'error' : 'info',
+  };
+
+  return (
+    <View
+      style={{
+        flex: 1,
+        marginBottom: 8,
+      }}
+    >
+      <TextInput
+        mode="flat"
+        disabled={disabled || readOnly}
+        multiline
+        error={!!error}
+        label={`${label as any}${(props as any).required ? ' *' : ''}`}
+        dense={true}
+        onChangeText={(text: string) => disabled || onChange(text)}
+        placeholder={placeholder}
+        ref={inputRef}
+        value={value}
+        style={{ backgroundColor: 'red' }}
+        underlineStyle={{
+          width: '90%',
+          height: 1.5,
+          backgroundColor: '#462A82',
+          bottom: 0,
+          left: '5%',
+       }}
+       activeUnderlineColor='#462A82'
+        {...(props as any)}
+      />
+      {!!formHelperText && (
+        <HelperText {...helperProps}>{formHelperText}</HelperText>
+      )}
+    </View>
+  );
+}
+
+export default connectField<MultiLineFieldProps>(Text, { kind: 'leaf' });
diff --git a/src/NumField.tsx b/src/NumField.tsx
index 21d49a4..a523646 100644
--- a/src/NumField.tsx
+++ b/src/NumField.tsx
@@ -1,6 +1,6 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { View } from 'react-native';
-import { FieldProps, connectField } from 'uniforms';
+import { FieldProps, connectField, useForm } from 'uniforms';
 import { TextInput, HelperText } from 'react-native-paper';

 export type NumFieldProps = FieldProps<
@@ -30,36 +30,88 @@ function Num({
   ...props
 }: NumFieldProps) {
   let formHelperText = (error && showInlineError && errorMessage) || helperText;
+  const [totalSum, setTotalSum] = useState(0);
+  const context = useForm();

   let helperProps: any = {
     type: showInlineError && error ? 'error' : 'info',
   };

+  useEffect(() => {
+    let values = context?.model;
+    let count = 0;
+    for (const key in values) {
+        const value = values[key];
+        if (typeof value === 'object' && value.hasOwnProperty('value')) {
+            count += value.value;
+            setTotalSum(count);
+        }
+    }
+  }, [context.model]);
+
+  useEffect(() => {
+    onChange(parseFloat(totalSum.toString()) as any)
+  }, [totalSum]);
+
   return (
+    props.field.function ?
     <View
-      style={{
-        flex: 1,
-        marginBottom: 8,
-      }}
-    >
-      <TextInput
-        keyboardType="numeric"
-        disabled={disabled || readOnly}
-        error={!!error}
-        label={`${label as any}${(props as any).required ? ' *' : ''}`}
-        dense={true}
-        onChangeText={(text: string) =>
-          disabled || onChange(parseFloat(text) as any)
-        }
-        placeholder={placeholder}
-        ref={inputRef}
-        value={!!value ? `${value}` : undefined}
-        {...(props as any)}
-      />
-      {!!formHelperText && (
-        <HelperText {...helperProps}>{formHelperText}</HelperText>
-      )}
-    </View>
+    style={{
+      flex: 1,
+      marginBottom: 8,
+    }}
+  >
+    <TextInput
+      mode="outlined"
+      keyboardType="numeric"
+      editable={false}
+      error={!!error}
+      label={`${label as any}${(props as any).required ? ' *' : ''}`}
+      dense={true}
+      placeholder={placeholder}
+      ref={inputRef}
+      value={!!totalSum ? `${totalSum}` : '0'}
+      onChangeText={() =>
+        disabled || onChange(parseFloat(totalSum.toString()) as any)
+      }
+      style={{ backgroundColor: 'transparent' }}
+      outlineColor='#462A82'
+      activeOutlineColor='#462A82'
+      {...(props as any)}
+    />
+    {!!formHelperText && (
+      <HelperText {...helperProps}>{formHelperText}</HelperText>
+    )}
+  </View>
+    :
+    <View
+    style={{
+      flex: 1,
+      marginBottom: 8,
+    }}
+  >
+    <TextInput
+      mode="outlined"
+      keyboardType="numeric"
+      disabled={disabled || readOnly}
+      error={!!error}
+      label={`${label as any}${(props as any).required ? ' *' : ''}`}
+      dense={true}
+      onChangeText={(text: string) =>
+        disabled || onChange(parseFloat(text) as any)
+      }
+      placeholder={placeholder}
+      ref={inputRef}
+      value={!!value ? `${value}` : undefined}
+      style={{ backgroundColor: 'transparent' }}
+      outlineColor='#462A82'
+      activeOutlineColor='#462A82'
+      {...(props as any)}
+    />
+    {!!formHelperText && (
+      <HelperText {...helperProps}>{formHelperText}</HelperText>
+    )}
+  </View>
   );
 }

diff --git a/src/RadioField.tsx b/src/RadioField.tsx
index c4b9891..43e497d 100644
--- a/src/RadioField.tsx
+++ b/src/RadioField.tsx
@@ -1,7 +1,9 @@
-import React from 'react';
-import { View } from 'react-native';
-import { FieldProps, connectField } from 'uniforms';
+import React, { useEffect, useState } from 'react';
+import { Platform, View } from 'react-native';
+import { FieldProps, connectField, useForm } from 'uniforms';
 import { RadioButton, Text, useTheme } from 'react-native-paper';
+import globalColors from '../../../../src/styles/globalColors';
+

 export type RadioFieldProps = FieldProps<any, any>;

@@ -17,43 +19,164 @@ function Radio({
   ...props
 }: RadioFieldProps) {
   let theme = useTheme();
+  const handleValueChange = (newValue: {}) => {
+    / if (!disabled && !readOnly) {
+    onChange(newValue);
+    / }
+  };

   return (
-    <View
-      style={{
-        flex: 1,
-        marginBottom: 8,
-      }}
-    >
-      <Text
+    props.field.sum ?
+      <View
         style={{
-          marginBottom: 8,
-          color: !!error ? theme.colors.error : theme.colors.text,
+          flex: 1,
         }}
       >
-        {`${label as any}${(props as any).required ? ' *' : ''}`}
-      </Text>
-      <RadioButton.Group
-        onValueChange={(value) => disabled || readOnly || onChange(value)}
-        value={value ?? ''}
+        <Text
+          style={{
+            marginTop: 8,
+            / marginBottom: 8,
+            fontSize: 20,
+            / backgroundColor: '#F2EFF8',
+            color: !!error ? theme.colors.error : theme.colors.text,
+          }}
+        >
+          {`${label as any}${(props as any).required ? ' *' : ''}`}
+        </Text>
+        <View
+            style={{
+              height: 1,
+              width: '100%',
+              alignSelf: 'center',
+              backgroundColor: globalColors.grey.medium,
+               marginTop:0,
+               marginBottom:20,
+               borderWidth: 0,
+               shadowOffset: {
+                  width: 1,
+                  height: 2,
+               },
+               shadowOpacity: 0.3,
+               shadowRadius: 2,
+               elevation: 2,
+            }}
+         />
+        <RadioButton.Group
+          onValueChange={handleValueChange}
+          value={value ?? ''}
+        >
+          {allowedValues?.map((item: any, idx: number) => (
+            <View
+              key={`${item}-${idx}`}
+              style={{
+                display: 'flex',
+                flexDirection: 'row',
+                justifyContent: 'flex-start',
+                alignContent: 'center',
+                alignItems: 'center',
+                marginBottom: 10
+              }}
+            >
+              {
+                Platform.OS === 'ios' && (
+                  <>
+                    <View style={{ borderWidth: 0.8, borderRadius: 50, borderColor: theme.dark ? globalColors.defaultLight : globalColors.defaultDark }}>
+                      <RadioButton value={item} color={theme.dark ? globalColors.defaultLight : globalColors.defaultDark} />
+                    </View>
+                    <View style={{ paddingRight: '5%' }}>
+                      <Text style={{ marginLeft: 15, marginRight: 15 }}>{transform ? transform(item.label) : item.label}</Text>
+                    </View>
+                  </>
+                )
+              }
+              {
+                Platform.OS === 'android' && (
+                  <>
+
+                    <RadioButton value={item} color={theme.dark ? globalColors.defaultLight : globalColors.defaultDark} />
+                    <View style={{ paddingRight: '5%' }}>
+                      <Text style={{ marginLeft: 15, marginRight: 15 }}>{transform ? transform(item.label) : item.label}</Text>
+                    </View>
+                  </>
+                )
+              }
+            </View>
+          ))}
+        </RadioButton.Group>
+      </View>
+      :
+      <View
+        style={{
+          flex: 1,
+        }}
       >
-        {allowedValues?.map((item: any, idx: number) => (
-          <View
-            key={`${item}-${idx}`}
+        <Text
+          style={{
+            marginTop: 8,
+            marginBottom: 8,
+            fontSize: 20,
+            / backgroundColor: '#F2EFF8',
+            color: !!error ? theme.colors.error : theme.colors.text,
+          }}
+        >
+          {`${label as any}${(props as any).required ? ' *' : ''}`}
+        </Text>
+        <View
             style={{
-              display: 'flex',
-              flexDirection: 'row',
-              justifyContent: 'flex-start',
-              alignContent: 'center',
-              alignItems: 'center',
+              height: 1,
+              width: '100%',
+              alignSelf: 'center',
+              backgroundColor: globalColors.grey.medium,
+               marginTop:0,
+               marginBottom:20,
+               borderWidth: 0,
+               shadowOffset: {
+                  width: 1,
+                  height: 2,
+               },
+               shadowOpacity: 0.3,
+               shadowRadius: 2,
+               elevation: 2,
             }}
-          >
-            <RadioButton value={item} />
-            <Text>{transform ? transform(item) : item}</Text>
-          </View>
-        ))}
-      </RadioButton.Group>
-    </View>
+         />
+        <RadioButton.Group
+          onValueChange={handleValueChange}
+          value={value ?? ''}
+        >
+          {allowedValues?.map((item: any, idx: number) => (
+            <View
+              key={`${item}-${idx}`}
+              style={{
+                display: 'flex',
+                flexDirection: 'row',
+                justifyContent: 'flex-start',
+                alignContent: 'center',
+                alignItems: 'center',
+                marginBottom: 10
+              }}
+            >
+              {
+                Platform.OS === 'ios' && (
+                  <>
+                    <View style={{ borderWidth: 0.8, borderRadius: 50, borderColor: theme.dark ? globalColors.defaultLight : globalColors.defaultDark }}>
+                      <RadioButton value={item} color={theme.dark ? globalColors.defaultLight : globalColors.defaultDark} />
+                    </View>
+                    <Text style={{ marginLeft: 15 }}>{transform ? transform(item) : item}</Text>
+                  </>
+                )
+              }
+              {
+                Platform.OS === 'android' && (
+                  <>
+                    <RadioButton value={item} color={theme.dark ? globalColors.defaultLight : globalColors.defaultDark} />
+                    <Text style={{ marginLeft: 15 }}>{transform ? transform(item) : item}</Text>
+                  </>
+                )
+              }
+            </View>
+          ))}
+        </RadioButton.Group>
+      </View>
   );
 }

diff --git a/src/RadioField2.tsx b/src/RadioField2.tsx
new file mode 100644
index 0000000..b3d275c
--- /dev/null
+++ b/src/RadioField2.tsx
@@ -0,0 +1,67 @@
+import React from 'react';
+import { Platform, View } from 'react-native';
+import { FieldProps, connectField } from 'uniforms';
+import { RadioButton, Text, useTheme } from 'react-native-paper';
+import globalColors from '../../../../src/styles/globalColors';
+import BouncyCheckbox from "react-native-bouncy-checkbox";
+export type RadioFieldProps = FieldProps<any, any>;
+
+function Radio({
+  error,
+  label,
+  value,
+  readOnly,
+  disabled,
+  transform,
+  allowedValues,
+  onChange,
+  ...props
+}: RadioFieldProps) {
+  let theme = useTheme();
+  return (
+    <View
+      style={{
+        flex: 1,
+        marginBottom: 8,
+      }}
+    >
+      <Text
+        style={{
+          marginBottom: 8,
+          color: !!error ? theme.colors.error : theme.colors.text,
+        }}
+      >
+        {`${label as any}${(props as any).required ? ' *' : ''}`}
+      </Text>
+
+        {allowedValues?.map((item: any, idx: number) => (
+            <View
+              key={`${item}-${idx}`}
+              style={{
+                display: 'flex',
+                flexDirection: 'row',
+                justifyContent: 'flex-start',
+                alignContent: 'center',
+                alignItems: 'center',
+                marginBottom: 10
+              }}
+            >
+               <BouncyCheckbox
+                size={99999}
+                fillColor="#9342f5"
+                unFillColor="#FFFFFF"
+                iconStyle={{ borderColor: "#9342f5", borderRadius: 50 }}
+                innerIconStyle={{ borderWidth: 2 }}
+                onPress={(value) => disabled || readOnly || onChange(value)}
+                text={transform ? transform(item) : item}
+                />
+            </View>
+
+        ))}
+
+    </View>
+  );
+}
+export default connectField<RadioFieldProps>(Radio, {
+  kind: 'leaf',
+});
diff --git a/src/TextAreaField.tsx b/src/TextAreaField.tsx
new file mode 100644
index 0000000..4792ed9
--- /dev/null
+++ b/src/TextAreaField.tsx
@@ -0,0 +1,72 @@
+import React from 'react';
+import { View } from 'react-native';
+import { FieldProps, connectField } from 'uniforms';
+import { TextInput, HelperText } from 'react-native-paper';
+
+export type TextAreaFieldProps = FieldProps<
+  string,
+  React.ComponentProps<typeof TextInput>,
+  {
+    helperText?: string;
+    inputRef?: any;
+  }
+>;
+
+function Text({
+  disabled,
+  error,
+  errorMessage,
+  helperText,
+  inputRef,
+  label,
+  name,
+  onChange,
+  placeholder,
+  readOnly,
+  showInlineError,
+  value = '',
+  ...props
+}: TextAreaFieldProps) {
+  let formHelperText = (error && showInlineError && errorMessage) || helperText;
+
+  let helperProps: any = {
+    type: showInlineError && error ? 'error' : 'info',
+  };
+
+  return (
+    <View
+      style={{
+        flex: 1,
+        marginBottom: 8,
+      }}
+    >
+      <TextInput
+        mode="flat"
+        disabled={disabled || readOnly}
+        multiline
+        error={!!error}
+        label={`${label as any}${(props as any).required ? ' *' : ''}`}
+        dense={true}
+        onChangeText={(text: string) => disabled || onChange(text)}
+        placeholder={placeholder}
+        ref={inputRef}
+        value={value}
+        style={{ backgroundColor: 'transparent' }}
+        underlineStyle={{
+          width: '90%',
+          height: 1.5,
+          backgroundColor: '#462A82',
+          bottom: 0,
+          left: '5%',
+       }}
+       activeUnderlineColor='#462A82'
+        {...(props as any)}
+      />
+      {!!formHelperText && (
+        <HelperText {...helperProps}>{formHelperText}</HelperText>
+      )}
+    </View>
+  );
+}
+
+export default connectField<TextAreaFieldProps>(Text, { kind: 'leaf' });
diff --git a/src/TextField.tsx b/src/TextField.tsx
index 7664a33..2d7b127 100644
--- a/src/TextField.tsx
+++ b/src/TextField.tsx
@@ -1,4 +1,4 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { View } from 'react-native';
 import { FieldProps, connectField } from 'uniforms';
 import { TextInput, HelperText } from 'react-native-paper';
@@ -34,6 +34,9 @@ function Text({
   };

   return (
+
+    props.field.multiline ?
+
     <View
       style={{
         flex: 1,
@@ -41,6 +44,8 @@ function Text({
       }}
     >
       <TextInput
+        mode="outlined"
+        multiline
         disabled={disabled || readOnly}
         error={!!error}
         label={`${label as any}${(props as any).required ? ' *' : ''}`}
@@ -49,12 +54,41 @@ function Text({
         placeholder={placeholder}
         ref={inputRef}
         value={value}
+        style={{ backgroundColor: 'transparent', minHeight: 100}}
+        outlineColor='#462A82'
+       activeOutlineColor='#462A82'
         {...(props as any)}
       />
       {!!formHelperText && (
         <HelperText {...helperProps}>{formHelperText}</HelperText>
       )}
     </View>
+    :
+    <View
+    style={{
+      flex: 1,
+      marginBottom: 8,
+    }}
+  >
+    <TextInput
+      mode="outlined"
+      disabled={disabled || readOnly}
+      error={!!error}
+      label={`${label as any}${(props as any).required ? ' *' : ''}`}
+      dense={true}
+      onChangeText={(text: string) => disabled || onChange(text)}
+      placeholder={placeholder}
+      ref={inputRef}
+      value={value}
+      style={{ backgroundColor: 'transparent' }}
+      outlineColor='#462A82'
+       activeOutlineColor='#462A82'
+      {...(props as any)}
+    />
+    {!!formHelperText && (
+      <HelperText {...helperProps}>{formHelperText}</HelperText>
+    )}
+  </View>
   );
 }

